memory_model/src/mmap.rs

```diff
@@ -371,9 +370,11 @@ mod tests {
     fn obj_read_and_write() {
         let mem_map = MemoryMapping::new(5).unwrap();
         assert!(mem_map.write_obj(55u16, 4).is_err());
+        assert!(mem_map.write_obj(55u16, core::usize::MAX).is_err());
```

Comment:
nit: ideally this would `match` the exact error instead of just checking it's **an** error.
Same for the other `err` tests added.

Reply:
thanks, I will try to improve it and post another version.

Reply:
Seems there's no easy way to compare enum by variants, and use a match clause will make the code a little complex. Do you prefer keeping "is_err()" or enhance it by using match clauses?
Thanks!